Sources/CXExtensions/DelayedAutoCancellable.swift

```diff
@@ -7,23 +7,26 @@ import CXFoundation
 extension Cancellable {
     
     public func cancel<S: Scheduler>(after interval: S.SchedulerTimeType.Stride, tolerance: S.SchedulerTimeType.Stride = .zero, scheduler: S, options: S.SchedulerOptions? = nil) -> DelayedAutoCancellable {
-        let cancel = DelayedAutoCancellable(self)
-        scheduler.schedule(after: scheduler.now.advanced(by: interval), tolerance: tolerance, options: options) {
-            cancel.cancel()
-        }
-        return cancel
+        return DelayedAutoCancellable(cancel: self, after: interval, tolerance: tolerance, scheduler: scheduler, options: options)
     }
 }
 
 public final class DelayedAutoCancellable: Cancellable {
     
-    private let c: Cancellable
+    private let canceller: Cancellable
+    
+    private var scheduleCanceller: Cancellable!
     
-    init<C>(_ cancel: C) where C: Cancellable {
-        self.c = cancel
+    public init<S: Scheduler>(cancel: Cancellable, after interval: S.SchedulerTimeType.Stride, tolerance: S.SchedulerTimeType.Stride, scheduler: S, options: S.SchedulerOptions? = nil) {
+        self.canceller = cancel
+        // FIXME: we should schedule non-repeatedly, but it's not cancellable.
+        self.scheduleCanceller = scheduler.schedule(after: scheduler.now.advanced(by: interval), interval: .seconds(.max), tolerance: tolerance, options: options) { [unowned self] in
+            self.cancel()
+        }
     }
     
     public func cancel() {
-        self.c.cancel()
+        scheduleCanceller.cancel()
+        canceller.cancel()
```

Comment:
I'm thinking if DelayedAutoCancellable should release these cancellers after cancel, just like [AnyCancellable](https://github.com/cx-org/CombineX/blob/95056aaa02a13caa55263ce44368633936af9a1a/Tests/CombineXTests/AnyCancellableSpec.swift#L67). 🤔

Reply:
Done. I'm wondering the type of `scheduleCanceller`. Should it be `Cancellable?` or `(() -> Void)?`

Reply:
`Cancellable` would make more sense, and why is `canceller` a closure in the new commit?